python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -652,3 +653,23 @@ def _isin_datetimelike(
 
     res = lhs._obtain_isin_result(rhs)
     return res
+
+
+def to_iso_calendar(arg):
+    formats = ["%G", "%V", "%u"]
+    if not isinstance(arg, (cudf.Index, cudf.core.series.DatetimeProperties)):
```

Comment:
@beckernick I'm not sure if this is the ideal way to catch anything that wouldn't work with `as_string_column` or `arg.series`.  Ideally, if something is either a series or index it should work with the rest of the code, but let me know if you think there'd be a better approach.